### DISCUSSION TITLE
Debugging timeout issues

### DISCUSSION THREAD

**Discussion by kaihendry** [serverless/serverless#10697] [Q&A]
I was first trying to find out why I should upgrade to version 3, but then I saw https://www.serverless.com/#How-It-works 's **Top-tier debugging tools, built right in.**

I have an issue whereby my function timeouts, but I don't know *where* it times out. Any pointers how to use these debug tools or if they can help in my case?

**Comment by mnapoli** [comment]
Hi!

Yes, the Serverless Dashboard will tell you if an execution resulted in a timeout, and then to further debug you can use the span view to see where your code stopped execution.

If the auto instrumented spans do not provide enough data you can also drop in your own spans with the Serverless Dashboard SDK.

Hope that helps!
